auth/gcloud/aio/auth/session.py

```diff
@@ -28,9 +29,11 @@
 
 class BaseSession:
     __metaclass__ = ABCMeta
+    Timeout = Union[int, float]
```

Comment:
Can we restrict the changes in this pr to the datastore library? Happy to consider another pr for updating types in the auth lib as well if that would be beneficial to you (there's at least one other typing fix we want to introduce anyway...)